cypress/plugins/index.js

```diff
@@ -55,5 +55,8 @@ module.exports = (on, config) => {
             // Pick correct set of values for given environment.
             const envSpecificConfig = content[environment] || {};
             return Object.assign(envSpecificConfig, unifiedCypressEnvVariables);
+        })
+        .then(envConfig => {
+            return require('@cypress/code-coverage/task')(on, { ...config, ...envConfig });
```

Comment:
This stuff is getting more and more crazy. :) But it is great that this worked.

I started to write out some possible changes, but since this seems to be working it seems better to leave it for now.

Hopefully at some point we can figure what this whole environment variable thing is doing.
One part that confused me is that at line 57 it is returning a config that has taken variables from `config.env` and moved them to the top level of config instead of putting them back in the `env` property.

Reply:
This may not be the best place to get into this, but the starting point for this is the `environments.json` file:
```json
{
  "production": {
    "baseUrl": "https://collaborative-learning.concord.org",
    "queryParams":"?appMode=qa&fakeClass=5&fakeUser=student:5&problem=2.1&qaGroup=5",
    "teacherQueryParams":"?appMode=qa&fakeClass=5&fakeUser=teacher:6&demoOffering=5&problem=2.1"
  },
  "master": {
    "baseUrl": "https://collaborative-learning.concord.org/branch/master/",
    "queryParams":"?appMode=qa&fakeClass=5&fakeUser=student:5&problem=2.1&qaGroup=5",
    "teacherQueryParams":"?appMode=qa&fakeClass=5&fakeUser=teacher:6&demoOffering=5&problem=2.1"
  },
  "branch": {
    "baseUrl": "https://collaborative-learning.concord.org/branch/problem-documents/",
    "queryParams":"?appMode=qa&fakeClass=5&fakeUser=student:5&problem=2.1&qaGroup=5",
    "teacherQueryParams":"?appMode=qa&fakeClass=5&fakeUser=teacher:6&demoOffering=5&problem=2.1"
  },
  "dev": {
    "baseUrl": "http://localhost:8080/",
    "queryParams":"?appMode=qa&fakeClass=5&fakeUser=student:5&demoOffering=5&problem=2.1&qaGroup=5",
    "teacherQueryParams":"?appMode=qa&fakeClass=5&fakeUser=teacher:6&demoOffering=5&problem=2.1"
  },
  "local":{
    "baseUrl": "http://localhost:8080/",
    "queryParams":"?appMode=qa&fakeClass=5&fakeUser=student:5&demoOffering=5&problem=2.1&qaGroup=5",
    "teacherQueryParams":"?appMode=qa&fakeClass=5&fakeUser=teacher:6&demoOffering=5&problem=2.1"
  }
}
```
So I believe the purpose of these machinations is to allow the environment to override certain top-level configuration parameters.

Reply:
Just for reference, I just followed this pattern from Piotr from the tectonics(?) repo. We should probably investigate if there is a better way of doing this or if we even need to do this.